topicExtractor.py

```diff
@@ -196,6 +198,34 @@ def fitTopicModel(self):
                 f"Failed to send message at max limit of {self.callMaxLimit} times."
             )
             return False
+        
+    
+    def useKMeans(self, vectorizerModel, docs, n_clusters=3):
+        """
+        Applies K-means clustering algorithm to group the given documents into topics.
+
+        Args:
+            vectorizerModel: The vectorizer model used to transform the documents into feature vectors.
+            docs: The list of documents to be clustered.
+            n_clusters: The number of clusters/topics to be created (default is 3).
+
+        Returns:
+            None
+        """
+
+        # This import should not occur frequently, so we only call when it is needed.
+        from sklearn.cluster import KMeans
+
+        # We use 3 clusters as a default value. 
+        # This means the transcript will be grouped into 3 topics.
+        clusterModel = KMeans(n_clusters=n_clusters)
+        try:
+            self.initializeTopicModel(vectorizerModel, clusterModel)
+            self.topics, probs = self.topicModel.fit_transform(docs)
+            return True
+        except:
+            logging.warn("Failed to fit the topic model with K-means clustering.")
+            return False
```

Comment:
So when the `vectorizerModel = None` it is throwing an error?

Reply:
No, I have not found a case where with no Vectorizermodel and KMeans as the clustering algorithm the clustering has failed. But I wrapped it in a try-catch to account for that possibility if it can happen.

Reply:
Thank you for pointing this out. I have added a log message for this purpose.

Reply:
I don't think you need two log message. Remove `logging.error` and incorporate the reason from the exception which is part of `e` into the message

`logging.warn(f"Failed to fit the topic model with K-means clustering due to {e}")`

Reply:
This has been addressed in the latest commit.

Reply:
This is the error it print in case of NO vector model case `Failed to fit the topic model with K-means clustering due **to Input contains infinity or a value too large for dtype('float64')**.`